File: Backend Java/ChatAppV2/src/main/java/com/chatapp/ChatAppV2/Models/Post.java
package com.chatapp.ChatAppV2.Models;

import java.time.LocalDate;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.stereotype.Component;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Component
public class Post {

    @Id
    private String id;

    private String username;

    private String displayName;

    private String postAccess; //public or private

    private String imageUrl;

    private int likes;

    private int numberOfComments;

    private List<Comment> comments;

    private String videoUrl;

    private String caption;

    private LocalDate postDate;

    private boolean liked;

    private String avatar;

    public boolean getLiked(){
        return liked;
    }

    public void setLiked(boolean liked){
        this.liked = liked;
    }

    
}

File: Frontend Kotlin/Talkative/app/src/main/java/com/example/talkative/model/OwnProfileResponse/Post.kt
package com.example.talkative.model.OwnProfileResponse




data class Post(
    val caption: String?,            // backend might send null
    val comments: List<String?>?,    // backend sends null instead of list
    val displayName: String?,        // nullable
    val id: String,
    val imageUrl: String?,
    val liked: Boolean,
    val likes: Int,
    val postAccess: String?,
    val postDate: String?,           // nullable
    val username: String?,           // nullable
    val videoUrl: String?,
    val avatar:String?
)
